File: Part8/C/FunctionConcepts/FunctionConcepts_7.c
// Example of function overloading
// C does not support function overloading, so this will not compile
#include <stdio.h>

void myFunction ( int IntParameter_Par )
{
    printf ( "Int version of my function called %d\n", IntParameter_Par );
}

void myFunction ( double DoubleParameter_Par )
{
    printf ( "Double version of my function called %.1f\n", DoubleParameter_Par );
}

int main ()
{
    myFunction ( 5 );
    myFunction ( 5.5 );
    return 0;
}

File: Part8/Rust/FunctionConcepts/FunctionConcepts_7.rs
// Example of function overloading
// Rust does not support function overloading, this will not compile
fn my_function ( int_parameter_par: i32 )
{
    println! ( "Int version of my function called {}", int_parameter_par );
}

fn my_function ( double_parameter_par: f64 )
{
    println! ( "Double version of my function called {:.1}", double_parameter_par );
}

fn main ()
{
    my_function ( 5 );
    my_function ( 5.5 );
}
